src/core/port-range-access.c

```diff
@@ -0,0 +1,271 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+/***
+  This file is part of systemd.
+
+  Copyright 2016 Daniel Mack
+  Copyright 2017 Intel Corporation.
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+***/
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "alloc-util.h"
+#include "bpf-firewall.h"
+#include "extract-word.h"
+#include "port-range-access.h"
+#include "parse-util.h"
+#include "string-util.h"
+
+static int create_series(uint32_t range_start, uint32_t range_end, PortRangeAccessItem **list) {
```

Comment:
ip ports are 16bit, we hence should use uint16_t whenever we handle them

Reply:
Ok, I'll do a uint16_t conversion everywhere.